homeassistant/components/esphome/__init__.py

```diff
@@ -215,12 +226,34 @@ def async_on_service_call(service: HomeassistantServiceCall) -> None:
                     **service_data,
                 },
             )
-        else:
+        elif entry.options.get(CONF_ALLOW_SERVICE_CALLS, DEFAULT_ALLOW_SERVICE_CALLS):
             hass.async_create_task(
                 hass.services.async_call(
                     domain, service_name, service_data, blocking=True
                 )
             )
+        else:
+            async_create_issue(
+                hass,
+                DOMAIN,
+                services_issue,
+                is_fixable=False,
+                severity=IssueSeverity.WARNING,
+                translation_key="service_calls_not_allowed",
+                translation_placeholders={
+                    "name": device_info.name,
```

Comment:
Should this be the friendly_name first?
```suggestion
                    "name": device_info.friendly_name or device_info.name,
```